app/components/post_list/post_list.js

```diff
@@ -96,25 +99,64 @@ export default class PostList extends PureComponent {
         mattermostManaged.removeEventListener(this.listenerId);
     }
 
+    handleClosePermalink = () => {
+        const {actions} = this.props;
+        actions.selectFocusedPostId('');
+        this.setState({showingPermalink: false});
+    };
+
+    handlePermalinkPress = (postId) => {
+        this.setState({isPermalink: true});
+        this.showPermalinkView(postId);
+    };
+
+    showPermalinkView = (postId) => {
+        const {actions, navigator} = this.props;
+        const {isPermalink, showingPermalink} = this.state;
```

Comment:
Maybe move `this.state.showingPermalink` to be just `this.showingPermalink` in case we get caught by state not being updated in time?